src/core/main.c

```diff
@@ -1782,6 +1783,24 @@ static int do_reexecute(
         assert(saved_rlimit_memlock);
         assert(ret_error_message);
 
+        if (switch_root_init) {
+                r = chase(switch_root_init, switch_root_dir, CHASE_PREFIX_ROOT, NULL, NULL);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to chase configured init %s%s: %m",
+                                          empty_to_root(switch_root_dir), switch_root_init);
```

Comment:
`switch_root_dir` may not end with `"/"`.
`"%s%s"` -> `"%s/%s"`
`empty_to_root(switch_root_dir)` -> `strempty(switch_root_dir)` ??